src/modules/common/common.providers.ts

```diff
@@ -15,4 +16,10 @@ export const commonProviders = [
       connection.model('Application', ApplicationSchema),
     inject: ['DATABASE_CONNECTION'],
   },
+  {
+    provide: 'MENTORSHIP_MODEL',
+    useFactory: (connection: Connection) =>
+      connection.model('Mentorship', MentorshipSchema),
+    inject: ['DATABASE_CONNECTION'],
+  },
```

Comment:
I'm not sure this is the correct way to make a model available for injection.  It's how the `User` model seems to be done, so I am following along.